src/org/tensorflow/yolo/view/ClassifierActivity.java

```diff
@@ -136,11 +184,65 @@ private void renderAdditionalInformation(final Canvas canvas) {
             }
         }
 
-        lines.add("Frame: " + previewWidth + "x" + previewHeight);
-        lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
-        lines.add("Rotation: " + sensorOrientation);
-        lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+// shimatani
+        // lines.add("Frame: " + previewWidth + "x" + previewHeight);
+        // lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
+        // lines.add("Rotation: " + sensorOrientation);
+        // lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+
+        Log.d(LOGGING_TAG, String.format("検出名: %s", nowRecognizedClass));
+        lines.add("検出名: " + nowRecognizedClass);
+
+        String msgInf = makeTts(nowRecognizedClass);
+
+        lines.add(msgInf1);
+        lines.add(msgInf2);
 
         borderedText.drawLines(canvas, 10, 10, lines);
     }
+
+    private void run() {
+        final long startTime = SystemClock.uptimeMillis();
+        final List<Recognition> results = recognizer.recognizeImage(croppedBitmap);
+        lastProcessingTimeMs = SystemClock.uptimeMillis() - startTime;
+        overlayView.setResults(results);
+
+        // shimatani
+        Log.d(LOGGING_TAG, "Debug: runInBackground()");
+
+        if (!(results.isEmpty())) {
+            nowResult = results.get(0).getTitle();
+            Log.d(LOGGING_TAG, String.format("Find: %s", nowResult));
+
+            if (!(results.isEmpty()) && lastResult.equals(nowResult)) {
+                matchCount++;
+                if (matchCount > 3) {
+                    // match 4 times
+                    matchCount = 0;
+
+                    if (!(results.isEmpty() || lastRecognizedClass.equals(nowResult))) {
+                        nowRecognizedClass = nowResult;
+                        Log.d(LOGGING_TAG, "Match 4 times: " + nowRecognizedClass);
+                        lastRecognizedClass = nowRecognizedClass;
+
+                        tts = makeTts(nowRecognizedClass);
+                        Log.d(LOGGING_TAG, "makeTts(): " + tts);
+                        if (!(tts.equals(""))) {
+                            speak2(results, tts);
```

Comment:
`speak2` method name looks very strange. Is there speak1, speak2, speak3...? What is the difference between the speak methods?

Reply:
I have prepared speak2 () function to speak two lines of message in Japanese.
I thought it would be better to change the name of the function.

TextToSpeechActivity.java
```TextToSpeechActivity.java
    protected void speak(List<Recognition> results) {
        if (!(results.isEmpty() || lastRecognizedClass.equals(results.get(0).getTitle()))) {
            lastRecognizedClass = results.get(0).getTitle();
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                textToSpeech.speak(lastRecognizedClass, TextToSpeech.QUEUE_FLUSH, null, null);
            } else {
                textToSpeech.speak(lastRecognizedClass, TextToSpeech.QUEUE_FLUSH, null);
            }
            Log.d(LOGGING_TAG, "speak: " + lastRecognizedClass);    // shimatani
        }
    }

    protected void speak2(List<Recognition> results, String tts) {
        this.tts = tts;
        Log.d(LOGGING_TAG, "speak2 entry: " + this.tts);
        if (!(results.isEmpty() || this.tts.equals(results.get(0).getTitle()))) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                textToSpeech.speak(this.tts, TextToSpeech.QUEUE_FLUSH, null, null);
            } else {
                textToSpeech.speak(this.tts, TextToSpeech.QUEUE_FLUSH, null);
            }
            Log.d(LOGGING_TAG, "speak2: " + this.tts);    // shimatani
        }
    }

```